regulation-worker/internal/delete/api/api.go

```diff
@@ -5,20 +5,77 @@ package api
 //called by delete/deleteSvc with (model.Job, model.Destination).
 //returns final status,error ({successful, failure}, err)
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
 
+	"github.com/rudderlabs/rudder-server/config"
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
 )
-type deleteManager interface {
-	Delete(ctx context.Context, job model.Job, destDetail model.Destination) (model.JobStatus, error)
-}
-type API struct {
-	DeleteManager deleteManager
-}
 
 //prepares payload based on (job,destDetail) & make an API call to transformer.
 //gets (status, failure_reason) which is converted to appropriate model.Error & returned to caller.
-func (api *API) Delete(ctx context.Context, job model.Job, destDetail model.Destination) (status model.JobStatus, err error) {
-	
-	return api.DeleteManager.Delete(ctx,job,destDetail)
+func Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus {
+
+	method := "DELETE"
+	endpoint := "/d-transformer/delete-users"
+	destTransformURL := config.GetEnv("DEST_TRANSFORM_URL", "http://localhost:9090")
+	url := fmt.Sprint(destTransformURL, endpoint)
+
+	bodySchema := mapJobToPayload(job, destName, destConfig)
+
+	reqBody, err := json.Marshal(bodySchema)
+	if err != nil {
+		return model.JobStatusFailed
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, url, bytes.NewReader(reqBody))
+	if err != nil {
+		return model.JobStatusFailed
+	}
+
+	client := &http.Client{}
```

Comment:
Avoid creating a new client every time
```suggestion
```

It is better to introduce a struct, convert the function to a struct method and have client there.